docs/plat/rpi3.rst

```diff
@@ -0,0 +1,297 @@
+ARM Trusted Firmware for Raspberry Pi 3
+=======================================
+
+.. section-numbering::
+    :suffix: .
+
+.. contents::
+
+The Raspberry Pi 3 is an inexpensive single-board computer that contains four
+Cortex-A53 cores, which makes it possible to have a port of the Arm Trusted
+Firmware.
+
+IMPORTANT NOTE: This port isn't secure. All of the memory used is DRAM, which is
+available from both the Non-secure and Secure worlds. This port shouldn't be
+considered more than a prototype to play with and implement elements like PSCI
+to support the Linux kernel.
+
+Design
+------
+
+The SoC used by the Raspberry Pi 3 is the Broadcom BCM2837. It is a strange SoC
+with a VideoCore IV that acts as primary processor (and loads everything from
+the SD card) and is located between all Arm cores and the DRAM.
+
+This explains why it is possible to change the execution state (AArch64/AArch32)
+depending on a few files on the SD card. We only care about the cases in which
+the cores boot in AArch64 mode.
+
+The rules are simple:
+
+- If a file called ``kernel8.img`` is located on the ``boot`` partition of the
+  SD card, it will load it and execute in EL2 in AArch64. Basically, it executes
+  a `default AArch64 stub`_ at address ``0x0``.
+
+- If there is also a file called ``armstub8.bin``, it will load it at address
+  ``0x0`` (instead of the default one) and execute it in EL3 in AArch64. All the
+  cores are powered on at the same time and start at address ``0x0``.
+
+This is ideal for us, as we can save the default AArch32 kernel in
+``kernel8.img`` and the Trusted Firmware and anything else we need in
+``armstub8.bin``, and forget about the default bootstrap code.
+
+However, we still want to load the kernel and have all cores available, which
+means that we need to make the secondary cores work in the way the kernel
+expects, as explained in `Secondary cores`_.
+
+Ideally, we want to boot the kernel in Hypervisor mode in AArch32. This means
+that BL33 can't be in EL2 in AArch64 mode. The architecture specifies that
+AArch32 Hypervisor mode isn't present when AArch64 is used for EL2.
```

Comment:
This doesn't seem relevant. This port seems to support either AArch32 or AArch64 depending on the flag RPI3_BL33_IN_AARCH32.